test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
We might consider changing the use of this to mock the `get-initialize-pool-instance-fn` and `initialize-scripting-container-fn` instead of `create-pool-instance` in jruby-utils so that the mocking is at least limited to functions defined within the immediate library being tested.